src/wallet/wallet.cpp

```diff
@@ -1722,20 +1722,22 @@ std::vector<uint256> CWallet::ResendWalletTransactionsBefore(int64_t nTime)
 {
     std::vector<uint256> result;
 
-    LOCK(cs_wallet);
-    // Sort them in chronological order
-    multimap<unsigned int, CWalletTx *> mapSorted;
-    for (PAIRTYPE(const uint256, CWalletTx) & item : mapWallet)
+    multimap<unsigned int, CWalletTx> mapSorted;
     {
-        CWalletTx &wtx = item.second;
-        // Don't rebroadcast if newer than nTime:
-        if (wtx.nTimeReceived > nTime)
-            continue;
-        mapSorted.insert(make_pair(wtx.nTimeReceived, &wtx));
+        LOCK(cs_wallet);
+        // Sort them in chronological order
```

Comment:
Ha, I also just fixed this in 1345 although differently, I guess you know it was a potential deadlock...I like you're fix better, will drop mine.